recipes/nvcomp/meta.yaml

```
@@ -0,0 +1,110 @@
+{% set version = "2.6.1" %}
+{% set cuda_major = (cuda_compiler_version|default("12.0.0")).split(".")[0] %}
+{% set platform = "x86_64" %}  # [linux64]
+{% set platform = "SBSA" %}  # [aarch64]
+{% set platform = "windows" %}  # [win]
+{% set extension = "tgz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: nvcomp
+  version: {{ version }}
+
+source:
+  - url: https://developer.download.nvidia.com/compute/nvcomp/{{ version }}/local_installers/nvcomp_{{ version }}_{{ platform }}_{{ cuda_major }}.x.{{ extension }}
+    sha256: 62046476076d1b455fb13fb78b38d3496bacafbfdbb0494fb3392240a04466ea  # [linux64 and cuda_major == "11"]
+    sha256: a0792b666caaf593a9dd4130979578fd3a78230f4407645c295700ef8e7aaaf2  # [linux64 and cuda_major == "12"]
+    sha256: a019f2c8a300991d94d718a3524377d978efadaa62349858741f17ef859a9e55  # [aarch64 and cuda_major == "11"]
+    sha256: fc668519a8344e9d05335bad4bc5d23a504cdc7579aea41f12d6aa0f3079e709  # [aarch64 and cuda_major == "12"]
+    sha256: 4f0105a51ac6ea947154e252c8b220baf3345803f55644b5aad84e22ae4f8986  # [win and cuda_major == "11"]
+    sha256: e362e1fbf6b101ad71aa5314f21d00a4353b90efbe5cdedc5fcfb0f3b68cd791  # [win and cuda_major == "12"]
+  # Include cudart license because it is statically linked.
+  - url: https://developer.download.nvidia.com/compute/cuda/redist/cuda_cudart/LICENSE.txt
+    fn: cudart_LICENSE
+    sha256: 5db25d4fd138013b563f9a3d1d87f7de7df1dac014fd4cccdfbb3435a5cff761
+
+build:
+  number: 0
+  # TODO: Enable aarch64 builds
+  # TODO: Enable Windows with CUDA 12 when it is supported
+  skip: True  # [win32 or osx or ppc64le or cuda_compiler_version in (undefined, "None", "10.2") or aarch64 or (win and cuda_major == "12")]
+  script:
+    - mkdir -pv $PREFIX/include              # [linux]
+    - mv -v include/* $PREFIX/include        # [linux]
+    - mkdir -pv $PREFIX/lib                  # [linux]
+    - mv -v lib/* $PREFIX/lib/               # [linux]
+    - mkdir %LIBRARY_INC%                                   # [win]
+    - copy %SRC_DIR%\\include\\* %LIBRARY_INC%\\            # [win]
+    - mkdir %LIBRARY_LIB%                                   # [win]
+    - copy %SRC_DIR%\\lib\\nvcomp*.lib %LIBRARY_LIB%\\      # [win]
+    - mkdir %LIBRARY_BIN%                                   # [win]
+    - copy %SRC_DIR%\\lib\\nvcomp*.dll %LIBRARY_BIN%\\      # [win]
+  ignore_run_exports_from:
+    - cudatoolkit
```

Think this is meant to ignore the CUDA compiler's `run_exports`, which will ignore `cudatoolkit` when relevant. Though please let me know if something else is intended here

```suggestion
    - {{ compiler('cuda') }}
```

---

Honestly I don't know if this is correct. I copied this from another recipe as it seems to be a common pattern. Do you know whether we need this at all?

---

Yeah think we want to change this as noted above as [`cudatoolkit` doesn't have a `run_export`]( https://github.com/conda-forge/cudatoolkit-feedstock/blob/531e4594992258568fe187bc5c4e40d8c9c57b27/recipe/meta.yaml#L535 ). Only `{{ compiler('cuda') }}` does (for [CUDA 11]( https://github.com/conda-forge/nvcc-feedstock/blob/04b48c9d3923aa71a3e021ddef7d04e6e12502c9/recipe/meta.yaml#L40-L46 ) & [CUDA 12]( https://github.com/conda-forge/cuda-nvcc-feedstock/blob/f602ed9c9566326d6f2effdfb0a3b80cff3661a7/recipe/meta.yaml#L44-L46 )). So think this change is what we want (for [example]( https://github.com/conda-forge/ucx-split-feedstock/blob/78e8d08b1391a6a29d05eb22bf8a30d1878954f2/recipe/meta.yaml#L23-L24 ))